src/dataflow/src/render/reduce.rs

```diff
@@ -1415,6 +1518,42 @@ where
                             (AggregateFunc::SumDecimal, AccumInner::SimpleNumber { accum, .. }) => {
                                 Datum::from(*accum)
                             }
+                            (
+                                AggregateFunc::SumAPD,
+                                AccumInner::APD {
+                                    accum,
+                                    pos_infs,
+                                    neg_infs,
+                                    nans,
+                                    non_nulls,
+                                },
+                            ) => {
+                                // Take a wide DecNum (accumulator) into a
+                                // narrow DecNum (datum). If the accumulator
+                                // exceeds the datum's min/max, this new value
+                                // will be +/- infinity. Naturally, if the
+                                // accumulator receives values that negate the
+                                // datum's overflow, this will start returning
+                                // finite values once more.
+                                let d = accum.0.to_width::<13>().unwrap();
+                                let inf_d = d.is_infinite();
+                                let neg_d = d.is_negative();
+                                let pos_inf = *pos_infs > 0 || (inf_d && !neg_d);
+                                let neg_inf = *neg_infs > 0 || (inf_d && neg_d);
+                                // NaNs include cases where we've seen a
+                                // mixture of positive and negative infinities.
+                                if *nans > 0 || (pos_inf && neg_inf) {
+                                    Datum::APD(OrderedDecimal(DecNum::<13>::nan()))
+                                } else if pos_inf {
+                                    Datum::APD(OrderedDecimal(DecNum::<13>::infinity()))
+                                } else if neg_inf {
+                                    Datum::APD(OrderedDecimal(-DecNum::<13>::infinity()))
+                                } else if *non_nulls == 0 {
```

Comment:
i can't tell if this branch is actually something that could be triggered

Reply:
It is, but is there something that could help clarify?

Reply:
so I guess i was confused because theres a lot of overlap between this branch and

```
1474                     let value = if accum.total > 0
1475                         && accum.inner.is_zero()
1476                         && aggr.func != AggregateFunc::Count
1477                     {
1478                         Datum::Null
```

the only case i see where non_nulls == 0, would be if total (and positive and negative infinity) all equalled zero - is where we would expect to hit this branch or is there some other case I'm missing